Bear.py: Add version if check_prerequisites fails

sasacocic opened the pull request:
allows the user to see what version of the software they
need to install.

Fixes https://github.com/coala/coala/issues/3982

arjunsinghy96 commented on coalib/bears/Bear.py:
```diff
@@ -375,6 +375,9 @@ def check_prerequisites(cls):
             if not requirement.is_installed():
+                package = (
+                    requirement.package if not requirement.version
+                    else requirement.package + ' ' + requirement)
```
I think it should be `requirement.package + ' ' + requirement.version`

sasacocic replied:
Done. Thanks 👍

arjunsinghy96 commented on coalib/bears/Bear.py:
```diff
@@ -375,6 +375,9 @@ def check_prerequisites(cls):
+                package = (
+                    requirement.package if not requirement.version
+                    else requirement.package + ' ' + requirement)
                 return requirement.package + ' is not installed. You can ' + (
```
`return package + ' is not installed .. `

sasacocic replied:
Done

arjunsinghy96 commented:
Can you add doctest for these changes. Everything else seems fine.

adhikasp commented:
Also [squash your commit](http://api.coala.io/en/latest/Developers/Git_Basics.html#squashing-your-commits) please 👍

supergr35 commented:
You also have a couple problems with your commit message. Please address them as our documentation tells you to after squashing your commits.

sasacocic commented:
I added a doctest and squashed my commit. I looked at the docs for writing good commits, but I wasn't sure what to do about my message could you please provide more detail.

RohanVB commented on coalib/bears/Bear.py:
```diff
@@ -370,12 +370,21 @@ def check_prerequisites(cls):
+                    requirement.package if not requirement.version
+                    else requirement.package + ' ' + requirement.version)
+                return package + ' is not installed. You can ' + (
                     'install it using ') + (
```
why not have it as `return package + ' is not installed. You can install it using ' +`

Shade5 replied:
I think it crosses the number of characters per line limit.

Shade5 commented:
@sasacocic Look at this for commit info http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html.
You should mention the file you have changed in the commit.

sasacocic commented:
The file that has been changed is included in the commit message in the short log. I looked at the doc for writing good commits and did my best to match it please let me know if there is anything else I can do.

supergr35 commented:
Quick thing in commit body: `Shows` -> `Show`

userzimmermann commented on coalib/bears/Bear.py:
```diff
@@ -370,12 +370,21 @@ def check_prerequisites(cls):
-                return requirement.package + ' is not installed. You can ' + (
+                package = (
+                    requirement.package if not requirement.version
+                    else requirement.package + ' ' + requirement.version)
```
IMO that logic should be moved to a `PackageRequirement.__str__` method so that we can just use `str(requirement)` here.

@sils @Adrianzatreanu Where is the `dependency_management` repo BTW? :)

Udayan12167 replied:
https://gitlab.com/coala/package_manager

sasacocic replied:
I'am confused since this is a different repository. I'm not exactly sure how to push this. Should I clone the package_manager repo and make changes to it or ...... further clarification about how to do this would be helpful thanks.

userzimmermann commented:
@sasacocic As @Siddhartha1234 mentioned the changes here are fine now. The rest needs to be done in https://gitlab.com/coala/package_manager before we can merge this PR. I created an issue for the new `PackageRequirement.__str__` method:

https://gitlab.com/coala/package_manager/issues/53

So if you want... Yes! Just clone that repository and create a merge request there :) Otherwise I will do it ;)

userzimmermann commented:
@sasacocic @Siddhartha1234 I just did the MR:

https://gitlab.com/coala/package_manager/merge_requests/33

@Adrianzatreanu Please approve the above and make a new `dependency_management` release to get this PR merged :)

userzimmermann commented:
@sils @jayvdb @Udayan12167 @yukiisbored We urgently need to merge https://gitlab.com/coala/package_manager/merge_requests/33 and create a new `dependency_management` release for this PR! And for some other stuff I want to contribute... :)

userzimmermann commented:
@sasacocic The new `dependency_management` release is on its way... :)

jayvdb commented:
The next release still has work to do, including MRs to review: https://gitlab.com/coala/package_manager/milestones/1

jayvdb commented on coalib/bears/Bear.py:
```diff
@@ -370,12 +370,18 @@ def check_prerequisites(cls):
         'really_bad_package is not installed. You can install it using ...'
 
+        >>> class anotherBear(Bear):
+        ...     REQUIREMENTS = {PipRequirement('bad_package','0.0.1')}
```
add space after comma, per pep8

jayvdb commented:
@sasacocic , the new release of dependency_management is done and merged.
Please rebase.  That should fix the problem.

jayvdb commented:
ack be285db

Mixih commented:
batch merging for the freeze

Mixih commented:
This pr is getting batch merged at  https://github.com/coala/coala/pull/4142 as it is release critical. Thanks for your contribution!
